src/util/ajax.js

```diff
@@ -85,7 +85,11 @@ exports.getJSON = function(requestParameters: RequestParameters, callback: Callb
             }
             callback(null, data);
         } else {
-            callback(new AJAXError(xhr.statusText, xhr.status, requestParameters.url));
+            if (xhr.status === 401) {
```

Comment:
What do you think about also testing the URL is on the `mapbox.com` domain? So long as we promise [compatibility with third party vector tile sources](https://www.mapbox.com/mapbox-gl-js/example/third-party/) we must be careful with logic that assumes we're talking to Mapbox servers.

Reply:
good idea!